mitodl/mitxonline
# refactor: remove EdX-Api-Key usage

### What are the relevant tickets?
[#2102](https://github.com/mitodl/mitxonline/issues/2102) 

### Description (What does it do?)
This PR removes EDX_API_KEY usage as Open edX is going to remove the EDX_API_KEY. Details can be found at https://github.com/openedx/edx-platform/issues/34039. It also fixes a potential impact on the [deactivate_enrollment](https://github.com/mitodl/edx-api-client/blob/master/edx_api/enrollments/__init__.py#L204) API for non-staff users if the EdX-Api-Key is removed.

### Screenshots (if appropriate):


### How can this be tested?

- Fetch, check out to this branch, install edx-api-client with editable mode having this [branch](https://github.com/mitodl/edx-api-client/tree/mudassir/remove-x-edx-api-key-usage), and make sure edx is configured.
- Test all the endpoints that currently utilize `EdX-Api-Key` for client requests. For example.
   - You can check the following tasks if they are working with open edX without issue.
   - openedx.tasks.change_edx_user_email_async
   - openedx.tasks.change_edx_user_name_async
   - openedx.tasks.create_edx_user_from_id
   - openedx.tasks.create_user_from_id
   - openedx.tasks.regenerate_openedx_auth_tokens
   - openedx.tasks.repair_faulty_openedx_users
   - openedx.tasks.retry_failed_edx_enrollments
   - openedx.tasks.update_edx_user_profile
   - courses.tasks.subscribe_edx_course_emails
   - courses.tasks.sync_courseruns_data
   You can also use management commands to test where applicable. i.e:
   - `regenerate_edx_auth_tokens`
   - `repair_missing_courseware_records`
   - `retry_edx_enrollment`
   - `sync_enrollments`
   - `update_edx_profiles`

Important to be tested:
- Test user enrollment with different course "privileged" modes, i.e., `audit`, and `verified`.
   - This is how I tested, But It would be good if the tester tests the different enrollment through front-end.
```
from edx_api.client import EdxApi
api = EdxApi({"access_token": "<ACCESS_TOKEN>"}, base_url="http://host.docker.internal:18000")
api.enrollments.create_student_enrollment(course_id="course-v1:edX+DemoX+Demo_Course", username="verified", mode="audit")
```
- Test the [deactivate_enrollment](https://github.com/mitodl/edx-api-client/blob/master/edx_api/enrollments/__init__.py#L204) API, here are the few mitxonline features/management commands I found that directly or indirectly use this API.
   - `unenroll_enrollment`
   - `refund_fulfilled_order`
   - `transfer_enrollment`
   or can test the course unenroll from user dashboard.


### Additional Context
- Related PR: https://github.com/mitodl/edx-api-client/pull/104.
